jupyter_server/gateway/gateway_client.py

```diff
@@ -12,12 +12,13 @@
 from abc import ABC, ABCMeta, abstractmethod
 from datetime import datetime, timezone
 from email.utils import parsedate_to_datetime
-from http.cookies import SimpleCookie
+from http.cookies import Morsel, SimpleCookie
 from socket import gaierror
 
 from jupyter_events import EventLogger
 from tornado import web
 from tornado.httpclient import AsyncHTTPClient, HTTPClientError, HTTPResponse
+from tornado.httputil import HTTPHeaders
```

Comment:
```suggestion
```
I *think* for the linting error, moving this import to the `if ty.TYPE_CHECKING` block might resolve it.